### ISSUE TITLE
NoSuchFileException when file names contain special characters

### ISSUE THREAD

**Issue by mauromol** [spring-projects/sts4#528]
Ok, I'm a bit silly, but since Java allows me to do it, why not? I have a Spring Data JPA repository named `IdentitàMittenteRepository` which works on `IdentitàMittenteEntity` entities.
When STS starts, I have exceptions like the following in the Spring Boot Language Server log:

```
11:29:52.685 [pool-2-thread-1] ERROR o.s.i.v.b.j.utils.SpringIndexerJava - error extracting symbol information in project 'fatturapa-service' - for docURI 'file:///home/mauro/svn/DCS-FATTURAPA-production/fatturapa-service/src/main/java/it/dcssrl/fatturapa/service/repository/IdentitàMittenteRepository.java' - on node: /** 
 * Returns the sender identity with a given name.
 * @param nome the identity name
 * @return the sender identity with the given name, or <code>null</code> ifno sender identity has that name
 */
IdentitàMittenteEntity findByNome(String nome);

java.nio.file.NoSuchFileException: /home/mauro/svn/myproject/mysubproject/src/main/java/it/dcssrl/myproject/service/repository/Identit�MittenteRepository.java
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:370)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:421)
	at java.base/java.nio.file.Files.readAllBytes(Files.java:3205)
	at org.springframework.ide.vscode.boot.java.utils.DocumentUtils.createTempTextDocument(DocumentUtils.java:36)
	at org.springframework.ide.vscode.boot.java.utils.DocumentUtils.getTempTextDocument(DocumentUtils.java:27)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.extractSymbolInformation(SpringIndexerJava.java:459)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.access$400(SpringIndexerJava.java:67)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava$3.visit(SpringIndexerJava.java:398)
	at org.eclipse.jdt.core.dom.MethodDeclaration.accept0(MethodDeclaration.java:593)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:2927)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:2998)
	at org.eclipse.jdt.core.dom.TypeDeclaration.accept0(TypeDeclaration.java:447)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:2927)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:2998)
	at org.eclipse.jdt.core.dom.CompilationUnit.accept0(CompilationUnit.java:258)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:2927)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.scanAST(SpringIndexerJava.java:381)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.access$000(SpringIndexerJava.java:67)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava$2.acceptAST(SpringIndexerJava.java:371)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:1068)
	at org.eclipse.jdt.core.dom.CompilationUnitResolver.resolve(CompilationUnitResolver.java:662)
	at org.eclipse.jdt.core.dom.ASTParser.createASTs(ASTParser.java:1013)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.scanFiles(SpringIndexerJava.java:375)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.scanFiles(SpringIndexerJava.java:327)
	at org.springframework.ide.vscode.boot.java.utils.SpringIndexerJava.initializeProject(SpringIndexerJava.java:113)
	at org.springframework.ide.vscode.boot.app.SpringSymbolIndex$InitializeProject.run(SpringSymbolIndex.java:645)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Please note that the file name on the file system is correct (`IdentitàMittenteRepository`), I'm working on Linux in a system with locale it_IT.UTF-8. No problems with Java compiler, Gradle, etc., however it seems like that, even if STS parses the source correctly, it fails at finding the right file name, by corrupting the accented letters (`Identit�MittenteRepository.java`). Perhaps it is handling file names with ISO-8859-1 or cp1252 charsets?

**Comment by martinlippert** [comment]
I can reproduce this, thanks a lot for raising this. This should indeed work.

Double checked this in VSCode and it seems to work just fine over there, so the next step here for me is to analyze where the URI for the doc with that special character is not encoded correctly. I am on it, but it will not make it into the upcoming (and slightly delayed) 4.7.2 release.

**Comment by martinlippert** [comment]
Fixed in the 4.8.0 release.
